File: UVA/vol-009/991.py
##  >>~~ UVa Online Judge ACM Problem Solution ~~<<

##  ID: 991
##  Name: Safe Salutations
##  Problem: https://onlinejudge.org/external/9/991.pdf
##  Language: Python

##  Author: Arash Shakery
##  Email: arash.shakery@gmail.com

DP = [1]
for i in range(1, 13):
    DP.append((4*i-2)*DP[i-1]//(i+1))

import sys
frst = 1
for line in sys.stdin:
    l = line.strip()
    if len(l) > 0:
        if not frst:
            print()
        else:
            frst = 0
        print(DP[int(line)])

File: UVA/vol-009/991.cpp
/*
  >>~~ UVa Online Judge ACM Problem Solution ~~<<

  ID: 991
  Name: Safe Salutations
  Problem: https://onlinejudge.org/external/9/991.pdf
  Language: C++

  Author: Arash Shakery
  Email: arash.shakery@gmail.com
*/

#include <bits/stdc++.h>
using namespace std;


int main() {
    int DP[13] = { 1 };
    for (int i=1; i<13; ++i)
        DP[i] = (4*i-2)*DP[i-1]/(i+1);

    int frst = 1, n;
    while (cin >> n) {
        if (frst) frst = 0;
        else cout << endl;
        cout << DP[n] << endl;
    }
}
